packages/core/src/domain/telemetry/telemetry.ts

```diff
@@ -87,7 +87,12 @@ export function startTelemetry(telemetryService: TelemetryService, configuration
         _dd: {
           format_version: 2 as const,
         },
-        telemetry: event as any, // https://github.com/microsoft/TypeScript/issues/48457
+        telemetry: combine(event, {
+          runtime_env: {
+            isLocalFile: isLocalFile(),
+            isWorker: isWorker(),
+          },
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** how about computing `runtime_env` once instead of for every event?